### PR TITLE
Clear conntrack entries on 0 -> 1 endpoint transition with externalIPs

### PR DESCRIPTION
**Pull request by JacobTanenbaum**
As part of the endpoint creation process when going from 0 -> 1 conntrack entries
are cleared. This is to prevent an existing conntrack entry from preventing traffic
to the service. Currently the system ignores the existance of the services external IP
addresses, which exposes that errant behavior

This adds the externalIP addresses of udp services to the list of conntrack entries that
get cleared. Allowing traffic to flow

Signed-off-by: Jacob Tanenbaum <jtanenba@redhat.com>

/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
UDP Service conntrack entries for ExternalIPs are now correctly cleared when endpoints are added 
```

### CODE REVIEW

**Comment by JacobTanenbaum** [comment]
@dcbw PTAL

**Comment by dcbw** [comment]
@thockin @freehan does this one look OK?

**Review by knobunc — APPROVED** [review]
/lgtm

@smarterclayton can you please approve this?

**Comment by spiffxp** [comment]
/hold
Can we get an explanation as to why this is important enough to introduce without an issue during code freeze?  What user-facing bug this fix?  Is this something that can wait until v1.15 or v1.14.1?

**Comment by JacobTanenbaum** [comment]
@spiffxp without this addition udp services with external IPs are broken if there is a stream of packets active when all endpoints go down and are restarted

**Comment by dcbw** [comment]
> What user-facing bug this fix? Is this something that can wait until v1.15 or v1.14.1?

@spiffxp if at all possible we'd like to get this fix into v1.14. If that's not possible, we might be OK with an /approve-for-1.14.1

**Comment by freehan** [comment]
LGTM

**Comment by freehan** [comment]
This PR https://github.com/kubernetes/kubernetes/pull/73323 should have covered the case right?

**Comment by JacobTanenbaum** [comment]
@freehan ideally it should have, PR https://github.com/kubernetes/kubernetes/pull/73323 cleans up conntrack entries for entries in endpointUpdateResult.StaleEndpoints which handles when an endpoint gets deleted. It does not account for clearing on endpoint creation (which only needs conntrack to be cleared when going from 0 -> 1 endpoints backing the service) which is accounted for in endpointUpdateResult.StaleServiceNames

**Comment by nikopen** [comment]
looks fairly important for me for stability of 1.14.0.

/approve

leaving the hold to @spiffxp

**Comment by dcbw** [comment]
@JacobTanenbaum thanks for the update that also fixes IPVS.
/lgtm

**Comment by spiffxp** [comment]
/hold cancel
thank you for the explanation and context all

**Comment by dcbw** [comment]
@spiffxp ok thanks! we'll work on getting some review and approval. @thockin can you take a look?

**Comment by xmudrii** [comment]
@JacobTanenbaum @thockin The [Code Thaw](https://github.com/kubernetes/sig-release/tree/master/releases/release-1.14#code-thaw) is starting tomorrow, Tuesday PST EOD. Would it be possible for this PR to get reviewed and approved by then? Otherwise it'll have to compete with bunch of other PRs post-thaw.

File: pkg/proxy/iptables/proxier.go
Comment (thockin) [inline pkg/proxy/iptables/proxier.go]: Should we catch status.ingress.ip too?

**Review by thockin — COMMENTED** [review]
Approving for velcoity but is this complete?  

/approve
/lgtm

File: pkg/proxy/iptables/proxier.go
Comment (JacobTanenbaum) [reply pkg/proxy/iptables/proxier.go]: @thockin yes we should catch status.ingress.ip also. I will submit another PR

**Pull request merged**
